blogpost/Blogpost-draft.md

```diff
@@ -1,39 +1,268 @@
+the main selling points of the new pipeline plugin. And we think that this is
+something we will see more of in the future. This can help solve two old problems
+with CI pipelines:
+* How to share job definitions for similar builds
+* How to version job definitions
+We think that having the job definition in a git repository and using the established
+tools and workflows to make changes to them addresses these problems. As this is still
+in its infancy we cannot yet tell if this is the sollution and what new problems this
+might produce. But we think it to be a step in the right direction.
+
+#### Location of the ``Jenkinsfile``
+
+It is a matter of personal style and preferences to decide whether the pipeline code (the ``Jenkinsfile``) should
+reside in the same repository as the actual project sources. The Jenkins developers and the community seem to advertise
+this style, which becomes quite visible when one tries to use the '[Multi-branch pipeline](#multibranch)' feature.
+After a lot of discussion, however, we agreed that we do not consider this a very good style, because:
+* It ties the project code together with Jenkins
+* When building 3rd-party code, it is required to fork it and add a Jenkinsfile
+* It puts information about the intended target environment into a build that should be agnostic to such information
+
+A possible way around this could be to use git's submodule feature. This would
+allow us to have a git repository with the ``Jenkins`` file and have the repository
+with the actual code added as a submodule. With this we would lose the advantage
+of Jenkins finding new branches and automatically creating a build pipeline for
+them but we could continue to keep our source code clean from Jenkins' configuration.
+As in most scenarios the number of parallel deployment environments is limited
+having multiple deployment pipelines might be not possible to beginn with.
+
+#### Missing things
+
+Even after working with pipelines for only a limited time it became clear that the whole
+DSL is still missing things you would expect to find. One such example is that most of
+the methods from the DSL have no return value you could use to find out information on
+your build. For example in our Docker pipeline we wanted to get the git commit ID to
+name our Docker container based on this. (Basically as an easy way to version our container
+and synchronize them with git.) To do we had to jump a few hoops:
+
+```groovy
+def gitCommitId
+stage("[COMMIT] checkout")
+node {
+  git branch: 'master', poll: true, url: 'https://github.com/Endron/dnd5-char-viewer.git'
+  sh 'git rev-parse HEAD > git.id'
+  gitCommitId = readFile('git.id')
+}
+```
+
+Because the ``git`` method has no return type we have to use git from the shell
+to get the commit ID. But because the ``sh`` method also has no return value we
+have to write the commit ID to a file we then read using the ``readFile`` method.
```

Comment:
The complete Git commit ID example doesn't belong to the conclusion - it should be moved to the earlier section about docker and just referenced here.